### PR TITLE
Add CIS Benchmark documentation to KubeOne

### PR DESCRIPTION
**Pull request by koksay**
This PR adds CIS Benchmark 1.8 documentation for KubeOne 1.7.3 and Kubernetes 1.27.

It especially requires a review for:
1. Formatting in general
2. `Details` section for `Not Applicable` controls
3. `Warn` and `Fail` controls, if they should be changed to `Not Applicable`

### CODE REVIEW

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (embik) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:543]: I think this should be Pass. We can still keep the details.

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (embik) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:156]: As a user, I don't understand why this is not applicable. Can we please elaborate what this is and why we don't think it's needed?

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (embik) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:144]: I'm not sure this is sufficient. At best, we'd disable anonymous auth completely. Do we have any idea why this isn't set by KubeOne/kubeadm?

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (embik) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:74]: What's the difference between Warn and Fail? Should we maybe explain that the beginning of the document?

**Review by embik — CHANGES_REQUESTED** [review]

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (koksay) [reply content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:74]: Actually, this is explained on top of the document, but I can update the wording.

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (koksay) [reply content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:543]: Ok, `not applicable` because of my script generation. I will update change it to `pass`

**Comment by koksay** [comment]
@embik I believe I covered all the items except `--anonymous-auth` parameter setting. To be honest, I have no idea why we don't set it at all (other than it does not cause issues because or our RBAC settings)

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (embik) [reply content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:74]: No, I just missed it for some reason. Sorry 🤦🏻

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (embik) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:236]: The URLs should be `ref` instead of absolute URL, so they link to the right version of the documentation. Here's an example: https://github.com/kubermatic/docs/blob/master/content/kubeone/main/_index.en.md?plain=1#L78

**Review by embik — CHANGES_REQUESTED** [review]

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (xmudrii) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:2]: We don't use hyphen to separate product name and version. Also, I'd consider rotating the Kubernetes version and the KubeOne version, and dropping the KubeOne patch release:
```suggestion
title = "Benchmark on Kubernetes 1.27 with KubeOne 1.7"
```

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (xmudrii) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:6]: Some minor nits:
```suggestion
This guide helps you evaluate the security of a Kubernetes cluster created using KubeOne against each control in the CIS Kubernetes Benchmark.
```

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (xmudrii) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:12]: Nit: please add the Kubernetes patch release that you used

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (xmudrii) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:20]:
```suggestion
🟢 **Pass:** The KubeOne cluster passes the audit/control outlined in the benchmark.
```

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (xmudrii) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:24]:
```suggestion
🔴 **Fail:** The audit/control will be fixed in a future KubeOne release.
```

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (xmudrii) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:186]: I was thinking more about this audit and I think it's actually a fail. We don't enable this admission control plugin, CIS requires it and doesn't go into details if it's alpha or not. It's up to us to decide if we want to support an alpha feature or not (we have always been towards not supporting alpha features and we can check with @csengerszabo if it's safe to state so in this document)

File: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md
Comment (xmudrii) [inline content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md:205]: When I think better, this is also a fail, for the same reason as the previous comment. We don't enable this admission control plugin and that's all what CIS cares about. We can say that this audit failed but that we don't support enabling this admission control plugin because the official Kubernetes docs state that it shouldn't be used at all and that it's deprecated.

**Review by xmudrii — CHANGES_REQUESTED** [review]
The term "KubeOne cluster" is used a lot through both documents but I'm not a huge fan of that term. I much more prefer using the term "Kubernetes cluster created using KubeOne" or something similar. "KubeOne cluster" sounds like a cluster of KubeOnes, but in reality it is a cluster of Kubernetes nodes. Other than that and other comments I left, this looks great to me!

**Comment by koksay** [comment]
@xmudrii Thanks again for the review. I updated the document based on your feedback.

I also added my doc generation python script to the `hack` directory, so it would be used later.

File: content/kubeone/main/guides/cis-benchmarking/_index.en.md
Comment (xmudrii) [inline content/kubeone/main/guides/cis-benchmarking/_index.en.md:8]: Nit:
```suggestion
In this document, information how it can be run on a Kubernetes cluster created using KubeOne and what to expect as the result is described.
```

File: hack/kube-bench-k1/master.json
Comment (xmudrii) [inline hack/kube-bench-k1/master.json:1]: Can we gitignore this file?

File: hack/kube-bench-k1/README.md
Comment (xmudrii) [inline hack/kube-bench-k1/README.md:5]: Nit:
```suggestion
kube-bench is deployed on the master node, and run this way:
```

**Review by xmudrii — CHANGES_REQUESTED** [review]
Some minor nits, after those are fixed, we can merge this PR and iterate later on if needed

**Comment by koksay** [comment]
@xmudrii I put it on the `main` documentation, should it also be under `1.7`?

**Comment by xmudrii** [comment]
@koksay Adding it to 1.7 makes sense, do you want to do that before we merge this PR?

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
